src/components/Image/ImageUtils.tsx

```diff
@@ -25,9 +25,9 @@ export function calculateDimensions({
 
   if (aspectRatio) {
     if (width) {
-      calculatedHeight = width / aspectRatio;
+      calculatedHeight = Math.round(width / aspectRatio);
     } else {
-      calculatedWidth = height * aspectRatio;
+      calculatedWidth = Math.round(height * aspectRatio);
```

Comment:
This is important to not get partial pixels. I checked and it seems most of the browsers round the numbers and not specifically floor/ceiling.